Fix links for Block, Chunk, ExecutionOutcome, Transaction, Shard, Receipt (#1856)
near/docs: docs/1.concepts/data-flow/near-data-flow.md

And a block appears on that timeline with some interval. Each of these blocks holds the information about the previous one, thus, creating a *chain of blocks*.

> NEAR Protocol has a sharded nature, meaning that more than one parallel network, called a [Shard](https://docs.near.org/develop/lake/structures/shard), can be live at any moment. And each Shard produces a chunk of a block at a given interval. A Block in NEAR Blockchain is a collection of the block chunks from all shards. Block chunk is shortened to [Chunk](https://near-indexers.io/docs/data-flow-and-structures/structures/chunk) in NEAR Protocol documentation.

Reviewer: ```suggestion
NEAR Protocol has a sharded nature, meaning that more than one parallel network, called a [Shard](https://docs.near.org/develop/lake/structures/shard), can be live at any moment. And each Shard produces a chunk of a block at a given interval. A Block in NEAR Blockchain is a collection of the block chunks from all shards. Block chunk is shortened to [Chunk](https://docs.near.org/develop/lake/structures/chunk) in NEAR Protocol documentation.
```

Returning to the data flow itself. The best we can do is to imagine tracks like we could see in audio/video editing apps. Each Shard has its own set of tracks. The top track is for Chunks. They appear no matter what at a given interval, for NEAR Blockchain the interval is about one second. Chunks are being produced even if nothing is happening on the blockchain.

![Timeline as tracks](/docs/flow/02-tracks.png)

> But what do we mean by saying something is happening? We mean that something has triggered some changes in the blockchain. And the most well-known method to trigger any changes is to send a [Transaction](https://docs.near.org/develop/lake/structures/protocol/transaction) to the blockchain with instructions of what we want to change and who has requested those changes.

Reviewer: ```suggestion
But what do we mean by saying something is happening? We mean that something has triggered some changes in the blockchain. And the most well-known method to trigger any changes is to send a [Transaction](https://docs.near.org/develop/lake/structures/transaction) to the blockchain with instructions of what we want to change and who has requested those changes.
```

> But what do we mean by saying something is happening? We mean that something has triggered some changes in the blockchain. And the most well known method to trigger any changes is to send a [Transaction](https://docs.near.org/concepts/protocol/transactions) to the blockchain with instructions of what we want to change and who has requested those changes.

Reviewer: ```suggestion
But what do we mean by saying something is happening? We mean that something has triggered some changes in the blockchain. And the most well-known method to trigger any changes is to send a [Transaction](https://docs.near.org/develop/lake/structures/protocol/transaction) to the blockchain with instructions of what we want to change and who has requested those changes.
```

A transaction needs to be constructed, signed and sent to the blockchain. Once it is executed we expect to have a result - [ExecutionOutcome](https://docs.near.org/develop/lake/structures/execution-outcome). Oh, that is simple. But it’s not really accurate for NEAR Blockchain.

![Transaction execution](/docs/flow/03-tx-outcome-receipt.png)

> In the beginning there is a [Transaction](https://docs.near.org/concepts/protocol/transactions), it contains instructions we want to be executed on the blockchain. Transaction is sent to NEAR Blockchain.

Reviewer: ```suggestion
In the beginning there is a [Transaction](https://docs.near.org/develop/lake/structures/transaction), it contains instructions we want to be executed on the blockchain. Transaction is sent to NEAR Blockchain.
```

And yes, it is executed there immediately BUT the immediate result of the transaction execution is always just *an acknowledgement that it will be executed on the chain;* this internal execution request is known as [Receipt](https://docs.near.org/develop/lake/structures/receipt). You can think of the [Receipt](https://docs.near.org/develop/lake/structures/receipt) as an internal transaction that exists to pass information across shards.

Let’s get back to the tracks to look at the example.

> Assuming we have two accounts living on different [Shards](https://docs.near.org/develop/lake/structures/shard) **alice.near** and **bob.near**. **alice.near** creates a [Transaction](https://docs.near.org/concepts/protocol/transactions) to send a few tokens to **bob.near**. The [Transaction](https://docs.near.org/concepts/protocol/transactions) is immediately executed and the [ExecutionOutcome](https://docs.near.org/develop/lake/structures/execution-outcome) for the [Transaction](https://docs.near.org/concepts/protocol/transactions) is always a [Receipt](https://docs.near.org/develop/lake/structures/receipt).

Reviewer: ```suggestion
Assuming we have two accounts living on different [Shards](https://docs.near.org/develop/lake/structures/shard) **alice.near** and **bob.near**. **alice.near** creates a [Transaction](https://docs.near.org/develop/lake/structures/transaction) to send a few tokens to **bob.near**. The [Transaction](https://docs.near.org/develop/lake/structures/transaction) is immediately executed and the [ExecutionOutcome](https://docs.near.org/develop/lake/structures/execution-outcome) for the [Transaction](https://docs.near.org/develop/lake/structures/transaction) is always a [Receipt](https://docs.near.org/develop/lake/structures/receipt).
```

## Summary

> We have learnt the main principles of how data is flowing in NEAR Protocol. We have found out that [Transactions](https://docs.near.org/concepts/protocol/transactions) execute and the [ExecutionOutcome](https://docs.near.org/develop/lake/structures/execution-outcome) of the Transaction is always a [Receipt](https://docs.near.org/develop/lake/structures/receipt).

Reviewer: ```suggestion
We have learnt the main principles of how data is flowing in NEAR Protocol. We have found out that [Transactions](https://docs.near.org/develop/lake/structures/transaction) execute and the [ExecutionOutcome](https://docs.near.org/develop/lake/structures/execution-outcome) of the Transaction is always a [Receipt](https://docs.near.org/develop/lake/structures/receipt).
```
